subsys/bluetooth/mesh/sensor_srv.c

```diff
@@ -801,12 +801,12 @@ static void pub_msg_add(struct bt_mesh_sensor_srv *srv,
 	uint16_t min_int = min_int_get(s, period_div, base_period);
 	int err;
 
-	if (srv->seq - s->state.seq < min_int) {
+	if ((uint16_t)(srv->seq - s->state.seq) < min_int) {
```

Comment:
Suggest storing this in a `uint16_t delta` on line 802 instead to avoid risking this bug in any future additional checks below

Reply:
nice find

Reply:
> nice find

Agreed! 🚀